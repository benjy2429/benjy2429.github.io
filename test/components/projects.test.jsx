import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Projects from '../../src/components/projects';

const mockProject = {
  title: 'Demo project',
  image: 'project-image.jpg',
  text: 'Project description'
};

const mockProjectWithLink = {
  ...mockProject,
  link: {
    text: 'Visit site',
    href: 'test.com'
  }
};

const mockProjectWithFile = {
  ...mockProject,
  file: 'file.pdf'
};

describe('Projects component', () => {
  it('renders projects', () => {
    const component = shallow(<Projects projects={[mockProject]} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders projects with a link', () => {
    const component = shallow(<Projects projects={[mockProjectWithLink]} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders projects with a file', () => {
    const component = shallow(<Projects projects={[mockProjectWithFile]} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
