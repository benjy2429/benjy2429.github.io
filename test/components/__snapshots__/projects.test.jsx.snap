// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component renders projects 1`] = `
<section
  className="projects"
>
  <div
    className="container"
  >
    <h2
      id="projects"
    >
      Projects
    </h2>
    <p>
      These are some of the projects I’ve worked on, both at university and in my spare time.
    </p>
    <div
      className="projects-wrapper"
    >
      <article
        className="project"
        key="Demo project"
      >
        <img
          alt=""
          className="project-image"
          src="/assets/projects/project-image.jpg"
        />
        <div
          className="project-content"
        >
          <h3
            className="project-title"
          >
            Demo project
          </h3>
          <p
            className="project-description"
          >
            Project description
          </p>
        </div>
      </article>
    </div>
  </div>
</section>
`;

exports[`Projects component renders projects with a file 1`] = `
<section
  className="projects"
>
  <div
    className="container"
  >
    <h2
      id="projects"
    >
      Projects
    </h2>
    <p>
      These are some of the projects I’ve worked on, both at university and in my spare time.
    </p>
    <div
      className="projects-wrapper"
    >
      <article
        className="project"
        key="Demo project"
      >
        <img
          alt=""
          className="project-image"
          src="/assets/projects/project-image.jpg"
        />
        <div
          className="project-content"
        >
          <h3
            className="project-title"
          >
            Demo project
          </h3>
          <p
            className="project-description"
          >
            Project description
          </p>
          <a
            className="project-link"
            href="/download/file.pdf"
          >
            Download
          </a>
        </div>
      </article>
    </div>
  </div>
</section>
`;

exports[`Projects component renders projects with a link 1`] = `
<section
  className="projects"
>
  <div
    className="container"
  >
    <h2
      id="projects"
    >
      Projects
    </h2>
    <p>
      These are some of the projects I’ve worked on, both at university and in my spare time.
    </p>
    <div
      className="projects-wrapper"
    >
      <article
        className="project"
        key="Demo project"
      >
        <img
          alt=""
          className="project-image"
          src="/assets/projects/project-image.jpg"
        />
        <div
          className="project-content"
        >
          <h3
            className="project-title"
          >
            Demo project
          </h3>
          <p
            className="project-description"
          >
            Project description
          </p>
          <a
            className="project-link"
            href="test.com"
          >
            Visit site
          </a>
        </div>
      </article>
    </div>
  </div>
</section>
`;
